src/test/credentials/provider/sharedCredentialsProvider.test.ts

```diff
@@ -0,0 +1,172 @@
+        )
+
+        assert.notStrictEqual(sut.validate(), undefined)
+        assertSubstringsInText(sut.validate(), MISSING_PROPERTIES_FRAGMENT, 'aws_access_key_id')
+    })
+
+    it('validation identifies when session_token is missing a corresponding secret key', async () => {
+        const sut = new SharedCredentialsProvider(
+            'default',
+            new Map<string, Profile>([['default', { aws_access_key_id: 'x', aws_session_token: 'z' }]])
+        )
+
+        assert.notStrictEqual(sut.validate(), undefined)
+        assertSubstringsInText(sut.validate(), MISSING_PROPERTIES_FRAGMENT, 'aws_secret_access_key')
+    })
+
+    it('validates a valid profile with an access key', async () => {
+        const sut = new SharedCredentialsProvider(
+            'default',
+            new Map<string, Profile>([['default', { aws_access_key_id: 'x', aws_secret_access_key: 'y' }]])
+        )
+
+        assert.strictEqual(sut.validate(), undefined)
+    })
+
+    it('validates a valid profile with a session token', async () => {
+        const sut = new SharedCredentialsProvider(
+            'default',
+            new Map<string, Profile>([
+                ['default', { aws_access_key_id: 'x', aws_secret_access_key: 'y', aws_session_token: 'z' }]
+            ])
+        )
+
+        assert.strictEqual(sut.validate(), undefined)
+    })
+
+    it('validates a valid profile with credential_process', async () => {
+        const sut = new SharedCredentialsProvider(
+            'default',
+            new Map<string, Profile>([['default', { credential_process: 'x' }]])
+        )
+
+        assert.strictEqual(sut.validate(), undefined)
+    })
+
+    it('validates a valid profile with role_arn', async () => {
+        const sut = new SharedCredentialsProvider(
+            'default',
+            new Map<string, Profile>([['default', { role_arn: 'x' }]])
+        )
+
+        assert.strictEqual(sut.validate(), undefined)
+    })
+
+    it('validates a valid profile with role_arn and source_profile', async () => {
+        const sut = new SharedCredentialsProvider(
+            'default',
+            new Map<string, Profile>([
+                ['default', { role_arn: 'x', source_profile: 'B' }],
+                ['B', { aws_access_key_id: 'x', aws_secret_access_key: 'y' }]
+            ])
+        )
+
+        assert.strictEqual(sut.validate(), undefined)
+    })
+
+    it('getCredentials throws when the profile is not valid', async () => {
+        const sut = new SharedCredentialsProvider(
+            'default',
+            new Map<string, Profile>([['default', { aws_access_key_id: 'x' }]])
+        )
+
+        await assertThrowsError(async () => {
+            await sut.getCredentials()
+        })
+    })
+})
+
+function assertSubstringsInText(text: string | undefined, ...substrings: string[]) {
+    assert.ok(text)
+    substrings.forEach(substring => assert.notStrictEqual(text!.indexOf(substring), -1))
+}
```

Comment:
Pull this out into the generic test utils.

Reply:
It makes sense to do that when there is a test that has a need for this to be shared

Reply:
I'm shocked that we don't have a single utils dir and (I don't think) any tests that check substrings. This stays.